src/components/PrivateHomeRoute.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import { Route, Redirect } from 'react-router-dom'
+import { getToken } from '../useToken'
+import PropTypes from 'prop-types'
+
+const PrivateHomeRoute = ({ component: Component, ...rest }) => {
+  return (
+        <Route {...rest} render={props => (
+          getToken() ? <Component {...props} /> : <Redirect to="/login" />
+        )} />
+  )
+}
+
+PrivateHomeRoute.propTypes = {
+  component: PropTypes.func.isRequired
+}
+
+export default PrivateHomeRoute
```

Comment:
Avoid a too short component like this.
- For the case `unauthenticated`, you can validate it in the App component by returning the login page if no token.
- For the case `authenticated` and go to log in, you can check in the login component by redirecting to `/home` if there's a token.